src/Formik.tsx

```diff
@@ -476,36 +471,55 @@ export function useFormik<Values = object>({
     [fields]
   );
 
-  const handleBlur = React.useCallback((eventOrString: any):
-    | void
-    | ((e: any) => void) => {
-    if (isString(eventOrString)) {
-      return event => executeBlur(event, eventOrString);
-    } else {
-      executeBlur(eventOrString);
-    }
+  const setTouched = React.useCallback(
+    (touched: FormikTouched<Values>) => {
+      dispatch({ type: 'SET_TOUCHED', payload: touched });
+      return validateOnBlur
+        ? validateFormWithLowPriority(state.values)
+        : Promise.resolve();
+    },
+    [validateFormWithLowPriority, state.values, validateOnBlur]
+  );
 
-    function executeBlur(e: any, path?: string) {
-      if (e.persist) {
-        e.persist();
-      }
-      const { name, id, outerHTML } = e.target;
-      const field = path ? path : name ? name : id;
-
-      if (!field && process.env.NODE_ENV !== 'production') {
-        warnAboutMissingIdentifier({
-          htmlContent: outerHTML,
-          documentationAnchorLink: 'handleblur-e-any--void',
-          handlerName: 'handleBlur',
-        });
-      }
+  const setErrors = React.useCallback((errors: FormikErrors<Values>) => {
+    dispatch({ type: 'SET_ERRORS', payload: errors });
+  }, []);
+
+  const setValues = React.useCallback(
+    (values: Values) => {
+      dispatch({ type: 'SET_VALUES', payload: values });
```

Comment:
in general, I like to avoid using magic string. Constants instead?

Reply:
We can probs switch to just an enum